src/app/util/attribute-table.cpp

```diff
@@ -238,6 +242,44 @@ EmberAfStatus emAfWriteAttribute(EndpointId endpoint, ClusterId cluster, Attribu
     // write the data unless this is only a test
     if (!justTest)
     {
+        // Check whether anything is actually changing, before we do any work
```

Comment:
This chunk begs to be factored out in an anonymous namespace function.

Reply:
Yep, will do.

Reply:
Done.